x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlQueryAction.java

```diff
@@ -110,6 +118,18 @@ private void doExecuteForked(Task task, EsqlQueryRequest request, ActionListener
         );
     }
 
+    private TimeValue timeout(EsqlQueryRequest request) {
+        TimeValue rTimeout = request.timeout();
+        if (rTimeout == null) {
+            return DEFAULT_TIMEOUT;
```

Comment:
I am not sure if we can have the default timeout at the request level here. Once we have the default, it will be very hard to remove as it's considered a breaking change. I think we need to discuss with the team. Let's remove it from this PR, then re-introduce it in a follow-up after the discussion.

Reply:
I think a default timeout could be a useful feature, but I'll move it to a Setting and I'll set it to `-1` (ie. no timeout) by default.
Let's discuss it with the team though, maybe there are better options.